crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -428,7 +428,8 @@ impl C8YHttpProxyActor {
         // If the provided url is c8y, add auth
         if self
             .end_point
-            .url_is_in_my_tenant_domain(download_info.url())
+            .maybe_tenant_url(download_info.url())
+            .is_some()
         {
             let token = self.get_and_set_jwt_token().await?;
             download_info.auth = Some(Auth::new_bearer(token.as_str()));
```

Comment:
This is out of scope for this PR, but we need to consider using `c8y_auth_proxy` here instead of managing jwt tokens.

@rina23q  and @Ruadhri17 You will have to consider using  `c8y_auth_proxy`  in your work on `tedge_configuration_plugin` (not in the plugin itself but for the urls forged by the c8y mapper for the plugin to download configuration files).